packages/valory/skills/decision_making_abci/tasks/staking.py

```diff
@@ -20,18 +20,59 @@
 """This package contains the logic for task preparations."""
 
 from datetime import datetime, timezone
-from typing import Generator, Optional
+from typing import Dict, Generator, List, Optional, Tuple, cast
 
 from packages.valory.contracts.staking.contract import Staking
 from packages.valory.protocols.contract_api import ContractApiMessage
 from packages.valory.skills.decision_making_abci.rounds import Event
 from packages.valory.skills.decision_making_abci.tasks.task_preparations import (
     TaskPreparation,
 )
-from packages.valory.skills.staking_abci.behaviours import (
-    BASE_CHAIN_ID,
-    get_activity_updates,
-)
+from packages.valory.skills.staking_abci.behaviours import BASE_CHAIN_ID
+
+
+POINTS_PER_ACTIVITY_UPDATE = 200
```

Comment:
This file contains the biggest changes. We now calculate the activity in this task. This calculation ended up being trickier than expected, as tweets need to be grouped to maximize the number of "updates" per user we send to the staking contact.

I basically sort tweets by points and keep adding them to the update as long as they produce a new "update", defined as 200 poitns currently.